### ISSUE TITLE
Blenderbim - Material Profile Set

### ISSUE THREAD

**Issue by brunopostle** [IfcOpenShell/IfcOpenShell#2052]
I'm trying to construct a composite profile using a Material Profile Set. It seems from the spec that each extrusion should be included in the `element.Representation.Representations` list.

But if I do this, blenderbim only displays the first Body Representation in this list (`IfcConvert` renders them all).

What does seem to work is to put them in the `element.Representations.Representation[0].Items` list, but this seems wrong.

Attached a sample IFC file that implements both, is one or the other or neither of them correct? 
[beam-typed.ifc.zip](https://github.com/IfcOpenShell/IfcOpenShell/files/8092652/beam-typed.ifc.zip)

**Comment by brunopostle** [comment]
> > It seems from the spec that each extrusion should be included in the element.Representation.Representations list.
> 
> Where did you get this conclusion?

It's probably an assumption I made (IfcConvert and ifcopenshell.validate seemed happy with it). If every item in the `element.Representation.Representations` list has to have a different Geometric Context, then this makes sense too.

> > What does seem to work is to put them in the element.Representation.Representations[0].Items list, but this seems wrong.
> 
> My interpretation is that this is correct.

This works, and is how I have been doing it in the Homemaker add-on, but it doesn't feel like the right solution, everything in this list has to have the same RepresentationType - how would you assemble a Window entirely out of extrusions for example?

@aothms I tried using a Shape Aspect to decorate a Beam Representation with another Representation, here: [beam-typed-2.ifc.zip](https://github.com/IfcOpenShell/IfcOpenShell/files/8095897/beam-typed-2.ifc.zip), but this decoration doesn't appear when I convert the model using IfcConvert (or in Blenderbim).

**Comment by aothms** [comment]
> but this decoration doesn't appear when I convert the model using IfcConvert 

IfcConvert doesn't have it implemented I think, but it shouldn't be too difficult. We can keep the issue open and remove the blenderbim label if that helps you.

**Comment by Moult** [comment]
> how would you assemble a Window entirely out of extrusions for example?

Potentially this is intentional, otherwise it expects modelers to be able to mix and match geometry paradigms within a single representation.

**Comment by brunopostle** [comment]
> > how would you assemble a Window entirely out of extrusions for example?
> 
> Potentially this is intentional, otherwise it expects modelers to be able to mix and match geometry paradigms within a single representation.

So, I know this is a question about Material Profile Sets, but say I wanted to create a Window Type to go in an IFC project library for reuse in blenderbim, and that Window has glass, wood and metal bits (some extruded, some brep etc..). This ultimately needs to be assembled using the Shape Aspect functionality?

**Comment by aothms** [comment]
The annoying thing is that the generally understood and supported mechanism is to use decomposition of the window into its various parts and use the normal mechanisms for associating material and property sets. However decomposition doesn't play that well probably with libraries and instantiations of types, BlenderBIM could support it (maybe it already does) but it'd have to instantiate not only the parent element but also all decomposing elements.  That's why the shape aspect thing exists so that you can encapsulate this all into a single product.

**Comment by Moult** [comment]
Yeah unfortunately instantiating aggregated types is not yet supported :(

**Comment by theoryshaw** [comment]
following

**Comment by Moult** [comment]
An update that shape aspects is now implemented. It only took us three years. In that time I've gotten married, bought a house, and had my first kid.
